[Win] Fix JSCOnly port compilation on Windows

bavulapati opened the pull request:
#### e784f928d3bf138cd11d2b42e437806768b71d12
<pre>
[Win] Fix JSCOnly port compilation on Windows
<a href="https://bugs.webkit.org/show_bug.cgi?id=259069">https://bugs.webkit.org/show_bug.cgi?id=259069</a>

Reviewed by NOBODY (OOPS!).

The JSCOnly build files currently miss few files to be included and few tweaks to configuration.
This tries to fix the same.

* Source/WTF/wtf/PlatformJSCOnly.cmake:
* Source/cmake/OptionsJSCOnly.cmake:
</pre><!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e784f928d3bf138cd11d2b42e437806768b71d12

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/35959 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/14904 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/38158 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/38681 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/32344 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/17320 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/11932 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/31087 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/36512 "Passed tests") | [❌ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/12600 "Found 1 new test failure: webrtc/audio-samplerate-change.html (failure)") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/31931 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/11061 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/11069 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/32128 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/39929 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/30437 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/32668 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/32449 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/37025 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/35784 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/11242 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/9158 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/35106 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/12988 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/42507 "Built successfully") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/11753 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/8826 "Passed tests") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/12073 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aperezdc commented on Source/WTF/wtf/PlatformJSCOnly.cmake:
```diff
@@ -1,3 +1,15 @@
+if (WTF_CPU_X86_64)
+    if (MSVC)
```
You could fold the two checks into a single `if`, as follows:

```suggestion
if (WTF_CPU_X86_64 AND MSVC)
```

donny-dont replied:
It was copy pasta from `PlatformWin.cmake` which should probably be doing that as well.

aperezdc reviewed (commented):
I'm not a Windows expert, but this touches only the CMake build system and the changes look reasonable, so `rs=me` but I would rather have someone with more Windows knowledge to do the final approval.

I left a comment about a possible simplification that would be nice to do before landing the patch, please take a look at that.

donny-dont commented on Source/cmake/OptionsJSCOnly.cmake:
```diff
@@ -1,4 +1,6 @@
-find_package(Threads REQUIRED)
+if (NOT WIN32)
+    find_package(Threads REQUIRED)
+endif ()
```
Why is this change needed? It should find the Windows library.

If its specifically due to something `vcpkg` is doing then this patch should live in that repository.

donny-dont commented on Source/WTF/wtf/PlatformJSCOnly.cmake:
```diff
@@ -161,6 +178,8 @@ else ()
+    list(APPEND WTF_LIBRARIES
+        Threads::Threads
+    )
+endif ()
```
See comment about `find_package(Threads)`

donny-dont commented on Source/cmake/OptionsJSCOnly.cmake:
```diff
@@ -55,10 +57,14 @@ endif ()
+      set(WTF_LIBRARY_TYPE OBJECT)
+    else ()
+      set(WTF_LIBRARY_TYPE SHARED)
+    endif ()
```
Can you just remove the `AND NOT WIN32` and the above FIXME? If it works that would be my preference.

bavulapati replied:
It is not fixed. We are reusing what `WinCairo` port is doing.
So we can not make it uniform across platforms for now.

donny-dont reviewed (changes requested):
Also as mentioned create a bug on https://bugs.webkit.org and use the `git-webkit pr` workflow to create a proper commit message.

foopoiuyt commented on Source/cmake/OptionsJSCOnly.cmake:
```diff
@@ -94,7 +100,10 @@ if (WIN32)
 
 string(TOLOWER ${EVENT_LOOP_TYPE} LOWERCASE_EVENT_LOOP_TYPE)
-if (LOWERCASE_EVENT_LOOP_TYPE STREQUAL "glib")
+if (WIN32)
```
Would it make more sense for this to be checked at the else rather than be first?

bavulapati replied:
If I understand it correctly, `glib` and the else case corresponds to *nix platforms. So having the `WIN32` condition first earlier made it easier order.

donny-dont replied:
@bavulapati its possible for someone to use glib on Windows so it makes sense to have that come first and the `elseif` be `WIN32`.

foopoiuyt commented on Source/WTF/wtf/PlatformJSCOnly.cmake:
```diff
@@ -122,7 +134,12 @@ else ()
 endif ()
 
-if (LOWERCASE_EVENT_LOOP_TYPE STREQUAL "glib")
+if (WIN32)
```
Similar to the other case, does it make more sense for this to be checked in the else of having the specific event loop type set?

bavulapati committed:
[Win] Fix JSCOnly port compilation on Windows
https://bugs.webkit.org/show_bug.cgi?id=259069

Reviewed by NOBODY (OOPS!).

The JSCOnly build files currently miss few files to be included and few tweaks to configuration.
This tries to fix the same.

* Source/WTF/wtf/PlatformJSCOnly.cmake:
* Source/cmake/OptionsJSCOnly.cmake:
